mobile_verifier/src/unique_connections/db.rs

```diff
@@ -0,0 +1,78 @@
+use std::ops::Range;
+
+use chrono::{DateTime, Utc};
+use file_store::unique_connections::UniqueConnectionsIngestReport;
+use futures::TryStreamExt;
+use helium_crypto::PublicKeyBinary;
+use sqlx::{FromRow, PgPool, Postgres, Transaction};
+
+use super::UniqueConnectionCounts;
+
+pub async fn get(
+    db: &PgPool,
+    reward_period: &Range<DateTime<Utc>>,
+) -> anyhow::Result<UniqueConnectionCounts> {
+    #[derive(FromRow)]
+    struct UniqueConnections {
+        hotspot_pubkey: PublicKeyBinary,
+        #[sqlx(try_from = "i64")]
+        unique_connections: u64,
+    }
+
+    let rows = sqlx::query_as::<_, UniqueConnections>(
+        r#"
+            SELECT hotspot_pubkey, unique_connections
+            FROM unique_connections
+            WHERE received_timestamp >= $1 AND received_timestamp <= $2
+            ORDER BY received_timestamp DESC
+            "#,
+    )
+    .bind(reward_period.start)
+    .bind(reward_period.end)
+    .fetch(db)
+    .and_then(|row| async move { Ok((row.hotspot_pubkey, row.unique_connections)) })
+    .try_collect()
+    .await?;
+
+    Ok(rows)
+}
+
+pub async fn save(
+    txn: &mut Transaction<'_, Postgres>,
+    report: &UniqueConnectionsIngestReport,
+) -> Result<(), sqlx::Error> {
+    // TODO: on conflict?
+    sqlx::query(
+        r#"
+        INSERT INTO unique_connections 
+        (hotspot_pubkey, unique_connections, start_timestamp, end_timestamp, received_timestamp)
+        VALUES
+        ($1, $2, $3, $4, $5)
```

Comment:
does this require an on conflict strategy ?

Reply:
The other side runs daily, so there should not be any conflicts I think but might still be interesting to add a do nothing. And also maybe some kind of index for the SELECT above?

Reply:
On conflict statement added. It only protects us from reprocessing the same file again.

The query for retrieving counts has a `DISTINCT ON` to make sure we only use the latest value provided for any radio.